[다오] step-2 체스판 생성

imojaek opened the pull request:
## 구현 내용
- 폰의 기본생성자를 만들어 매개인자가 없어도 하얀색 폰이 생성될 수 있게 했습니다.
- 보드 클래스를 만들고, 8*8 배열의 Pawn 배열 board를 선언했습니다. 이후 Piece 등의 배열로 변경될 것 같습니다.
- board에 폰을 추가하고, 이후 findPawn에서 사용하기 위한 리스트에 추가하는 add(Pawn),
- Pawn 배열에서 값이 들어있는지 확인하고, 그 개수를 반환하는 size(),
- 폰이 보드에 추가된 순서에 따라 Pawn을 반환하는 findPawn(Pawn)을 추가했습니다. 

## 고민 사항
- 


## 기타
- feature1 PR를 보낸 후에 바로feature2 브랜치를 만들어 작성했습니다. feature1에 수정사항이 생겼고, feature1에 `수정사항을 반영한 뒤` merge되었습니다. 
- 수정사항이 반영되기 전의 feature1에서 뻗어 나온 feature2, 그리고 수정사항이 반영된 feature1을 rebase를 써서 병합하려는 시도를 했으나 실패하고 feature2를 다시 작성했습니다. 다시 이런 일이 발생한다면 merge를 사용해보려고 합니다.

imojaek committed:
.gitignore 에 .idea 폴더 추가

인텔리제이에서 생성되는 파일이 들어가는 .idea폴더를 추가했습니다.

imojaek committed:
Pawn의 기본생성자 추가

폰의 기본생성자를 추가하고, 테스트코드를 작성했습니다.
그리고 폰의 색상 2가지에 대해 폰 내부의 필드로 상수인 WHITE, BLACK을 생성했습니다.

imojaek committed:
Pawn의 색상 상수를 public으로 변경

보드테스트 코드에서 Pawn.WHITE_COLOR 로 색상에 접근하기 위해서 private 였던 필드를 public으로 변경했습니다.

imojaek committed:
Board 클래스의 add(Pawn), size() 추가

add(Pawn pawn) : board 배열의 좌상단부터 순회하며 비어있는 공간을 찾고(findFreeBoard 메소드), 그 위치에 pawn을 넣습니다.

size() : board에 들어있는 Pawn의 개수를 세서 반환하는 메소드입니다.

imojaek committed:
pieces 패키지 chess 하위로 이동

imojaek committed:
index를 통해 폰을 찾는 findPawn 메소드 작성

매개로 받는 index를 이용해서 폰을 반환하는 findPawn 메소드를 작성했습니다.
인덱스는 폰이 추가된 순서처럼 보여, add 메소드를 통해 폰을 등록할 때마다 Board 내부의 pawnList에 추가하여 순서를 유지했습니다.

imojaek committed:
app\.idea 폴더 삭제

imojaek committed:
테스트코드의 상수 필드로 변경

craete() 테스트 클래스 내부에 선언되어있던 WHITE, BLACK 상수를 create_기본생성자 테스트코드에서도 사용하기 위해 필드로 변경했습니다.

honux77 commented on app/.idea/.gitignore:
요거는 삭제하는 게 아니라 gitignore에 추가해야 해요. 나중에 또 생깁니다.
지금은 삭제가 커밋된 상태군요.

imojaek replied:
제가 기존에 .idea 폴더를 .gitignore에 추가하지 않아 origin에 .idea 폴더가 올라가 있었습니다.
그래서 .gitignore의 마지막에

#IntelliJ directory
.idea

를 추가한 뒤에 origin에 올라가있는 .idea 폴더만 삭제하고, 로컬의 .idea는 남겨두었습니다.

honux77 reviewed (changes requested):
체스 보드를 2차원 배열로 만들었는데 지금의 요구사항에서는 아직 필요없는 내용이에요. 
일단 테스트 코드를 통과할 정도로 간결하게 줄여 주시면 좋겠습니다.

imojaek committed:
수정 : 요구사항에서 벗어나는 코드 제거

기존에 8*8 배열의 board를 선언하고, 빈자리를 찾아 폰을 추가하는 메소드를
작성했었습니다

이를 제거하고 요구사항의 내용과 테스트코드를 처리할 수 있을만큼의 코드만
남기도록 했습니다.

imojaek commented:
> 체스 보드를 2차원 배열로 만들었는데 지금의 요구사항에서는 아직 필요없는 내용이에요. 일단 테스트 코드를 통과할 정도로 간결하게 줄여 주시면 좋겠습니다.

2차원 배열을 선언함으로 인해 발생한 추가적인 코드들을 전부 삭제 후 개선하고, 테스트코드의 동작을 확인했습니다.

이후 미션에서 요구사항을 벗어나지 않도록 신경써보겠습니다.

honux77 reviewed (approved):
수정하느라 고생하셨어요. 👏
